### ISSUE TITLE
[boost-cmake] Build error on x64-windows

### ISSUE THREAD

**Issue by Mrguanglei** [microsoft/vcpkg#39994]
Computing installation plan...
The following packages will be built and installed:
    boost:x64-windows@1.85.0#1
  * boost-accumulators:x64-windows@1.85.0#1
  * boost-algorithm:x64-windows@1.85.0#1
  * boost-align:x64-windows@1.85.0#1
  * boost-any:x64-windows@1.85.0#1
  * boost-array:x64-windows@1.85.0#1
  * boost-asio:x64-windows@1.85.0#2
  * boost-assert:x64-windows@1.85.0#1
  * boost-assign:x64-windows@1.85.0#1
  * boost-atomic:x64-windows@1.85.0#1
  * boost-beast:x64-windows@1.85.0#2
  * boost-bimap:x64-windows@1.85.0#2
  * boost-bind:x64-windows@1.85.0#1
  * boost-callable-traits:x64-windows@1.85.0#1
  * boost-charconv:x64-windows@1.85.0#1
  * boost-chrono:x64-windows@1.85.0#1
  * boost-circular-buffer:x64-windows@1.85.0#1
  * boost-cmake:x64-windows@1.85.0#1
  * boost-cobalt:x64-windows@1.85.0#2
  * boost-compat:x64-windows@1.85.0#2
  * boost-compatibility:x64-windows@1.85.0#1
  * boost-compute:x64-windows@1.85.0#1
  * boost-concept-check:x64-windows@1.85.0#1
  * boost-config:x64-windows@1.85.0#1
  * boost-container:x64-windows@1.85.0#1
  * boost-container-hash:x64-windows@1.85.0#1
  * boost-context:x64-windows@1.85.0#1
  * boost-contract:x64-windows@1.85.0#1
  * boost-conversion:x64-windows@1.85.0#1
  * boost-convert:x64-windows@1.85.0#1
  * boost-core:x64-windows@1.85.0#2
  * boost-coroutine:x64-windows@1.85.0#1
  * boost-coroutine2:x64-windows@1.85.0#1
  * boost-crc:x64-windows@1.85.0#1
  * boost-date-time:x64-windows@1.85.0#1
  * boost-describe:x64-windows@1.85.0#2
  * boost-detail:x64-windows@1.85.0#1
  * boost-dll:x64-windows@1.85.0#1
  * boost-dynamic-bitset:x64-windows@1.85.0#1
  * boost-endian:x64-windows@1.85.0#1
  * boost-exception:x64-windows@1.85.0#1
  * boost-fiber:x64-windows@1.85.0#1
  * boost-filesystem:x64-windows@1.85.0#2
  * boost-flyweight:x64-windows@1.85.0#1
  * boost-foreach:x64-windows@1.85.0#1
  * boost-format:x64-windows@1.85.0#1
  * boost-function:x64-windows@1.85.0#1
  * boost-function-types:x64-windows@1.85.0#1
  * boost-functional:x64-windows@1.85.0#1
  * boost-fusion:x64-windows@1.85.0#1
  * boost-geometry:x64-windows@1.85.0#2
  * boost-gil:x64-windows@1.85.0#1
  * boost-graph:x64-windows@1.85.0#1
  * boost-hana:x64-windows@1.85.0#1
  * boost-headers:x64-windows@1.85.0#1
  * boost-heap:x64-windows@1.85.0#1
  * boost-histogram:x64-windows@1.85.0#1
  * boost-hof:x64-windows@1.85.0#1
  * boost-icl:x64-windows@1.85.0#1
  * boost-integer:x64-windows@1.85.0#1
  * boost-interprocess:x64-windows@1.85.0#1
  * boost-interval:x64-windows@1.85.0#1
  * boost-intrusive:x64-windows@1.85.0#2
  * boost-io:x64-windows@1.85.0#1
  * boost-iostreams[bzip2,core,lzma,zlib,zstd]:x64-windows@1.85.0#1
  * boost-iterator:x64-windows@1.85.0#1
  * boost-json:x64-windows@1.85.0#2
  * boost-lambda:x64-windows@1.85.0#1
  * boost-lambda2:x64-windows@1.85.0#1
  * boost-leaf:x64-windows@1.85.0#1
  * boost-lexical-cast:x64-windows@1.85.0#1
  * boost-local-function:x64-windows@1.85.0#1
  * boost-locale:x64-windows@1.85.0#2
  * boost-lockfree:x64-windows@1.85.0#1
  * boost-log:x64-windows@1.85.0#2
  * boost-logic:x64-windows@1.85.0#1
  * boost-math:x64-windows@1.85.0#2
  * boost-metaparse:x64-windows@1.85.0#1
  * boost-move:x64-windows@1.85.0#1
  * boost-mp11:x64-windows@1.85.0#1
  * boost-mpl:x64-windows@1.85.0#1
  * boost-msm:x64-windows@1.85.0#2
  * boost-multi-array:x64-windows@1.85.0#1
  * boost-multi-index:x64-windows@1.85.0#2
  * boost-multiprecision:x64-windows@1.85.0#2
  * boost-mysql:x64-windows@1.85.0#2
  * boost-nowide:x64-windows@1.85.0#2
  * boost-numeric-conversion:x64-windows@1.85.0#1
  * boost-odeint:x64-windows@1.85.0#1
  * boost-optional:x64-windows@1.85.0#1
  * boost-outcome:x64-windows@1.85.0#2
  * boost-parameter:x64-windows@1.85.0#1
  * boost-parameter-python:x64-windows@1.85.0#1
  * boost-pfr:x64-windows@1.85.0#2
  * boost-phoenix:x64-windows@1.85.0#1
  * boost-poly-collection:x64-windows@1.85.0#1
  * boost-polygon:x64-windows@1.85.0#1
  * boost-pool:x64-windows@1.85.0#1
  * boost-predef:x64-windows@1.85.0#1
  * boost-preprocessor:x64-windows@1.85.0#1
  * boost-process:x64-windows@1.85.0#2
  * boost-program-options:x64-windows@1.85.0#2
  * boost-property-map:x64-windows@1.85.0#1
  * boost-property-tree:x64-windows@1.85.0#1
  * boost-proto:x64-windows@1.85.0#1
  * boost-ptr-container:x64-windows@1.85.0#1
  * boost-python[core,python3]:x64-windows@1.85.0#1
  * boost-qvm:x64-windows@1.85.0#1
  * boost-random:x64-windows@1.85.0#1
  * boost-range:x64-windows@1.85.0#1
  * boost-ratio:x64-windows@1.85.0#1
  * boost-rational:x64-windows@1.85.0#1
  * boost-redis:x64-windows@1.85.0#2
  * boost-regex:x64-windows@1.85.0#1
  * boost-safe-numerics:x64-windows@1.85.0#1
  * boost-scope:x64-windows@1.85.0#1
  * boost-scope-exit:x64-windows@1.85.0#1
  * boost-serialization:x64-windows@1.85.0#1
  * boost-signals2:x64-windows@1.85.0#1
  * boost-smart-ptr:x64-windows@1.85.0#1
  * boost-sort:x64-windows@1.85.0#1
  * boost-spirit:x64-windows@1.85.0#1
  * boost-stacktrace:x64-windows@1.85.0#2
  * boost-statechart:x64-windows@1.85.0#1
  * boost-static-assert:x64-windows@1.85.0#1
  * boost-static-string:x64-windows@1.85.0#1
  * boost-stl-interfaces:x64-windows@1.85.0#1
  * boost-system:x64-windows@1.85.0#1
  * boost-test:x64-windows@1.85.0#1
  * boost-thread:x64-windows@1.85.0#1
  * boost-throw-exception:x64-windows@1.85.0#1
  * boost-timer:x64-windows@1.85.0#2
  * boost-tokenizer:x64-windows@1.85.0#1
  * boost-tti:x64-windows@1.85.0#1
  * boost-tuple:x64-windows@1.85.0#1
  * boost-type-erasure:x64-windows@1.85.0#1
  * boost-type-index:x64-windows@1.85.0#1
  * boost-type-traits:x64-windows@1.85.0#1
  * boost-typeof:x64-windows@1.85.0#1
  * boost-ublas:x64-windows@1.85.0#1
  * boost-units:x64-windows@1.85.0#1
  * boost-unordered:x64-windows@1.85.0#2
  * boost-url:x64-windows@1.85.0#1
  * boost-utility:x64-windows@1.85.0#1
  * boost-uuid:x64-windows@1.85.0#1
  * boost-variant:x64-windows@1.85.0#1
  * boost-variant2:x64-windows@1.85.0#2
  * boost-vmd:x64-windows@1.85.0#1
  * boost-wave:x64-windows@1.85.0#2
  * boost-winapi:x64-windows@1.85.0#1
  * boost-xpressive:x64-windows@1.85.0#1
  * boost-yap:x64-windows@1.85.0#1
  * bzip2[core,tool]:x64-windows@1.0.8#5
  * expat:x64-windows@2.6.2#1
  * libffi:x64-windows@3.4.6
  * liblzma:x64-windows@5.6.2
  * openssl:x64-windows@3.3.1#1
  * pkgconf:x64-windows@2.2.0
  * python3:x64-windows@3.11.8#4
  * sqlite3[core,json1]:x64-windows@3.46.0#2
  * vcpkg-cmake-get-vars:x64-windows@2023-12-31
  * vcpkg-get-python:x64-windows@2024-06-08
  * vcpkg-msbuild:x64-windows@2023-08-08
  * vcpkg-pkgconfig-get-modules:x64-windows@2024-04-03
  * vcpkg-tool-meson:x64-windows@1.3.2#8
  * zlib:x64-windows@1.3.1
  * zstd:x64-windows@1.5.6
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:10809".
Compiler found: E:/Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\20386\AppData\Local\vcpkg\archives in 2.65 ms. Use --debug to see more details.
Installing 1/167 boost-cmake:x64-windows@1.85.0#1...
Building boost-cmake:x64-windows@1.85.0#1...
-- Downloading https://github.com/boostorg/boost/archive/boost-1.85.0.tar.gz -> boostorg-boost-boost-1.85.0.tar.gz...

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):

      Failed to download file with error: no such file or directory
      If you are using a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.

      3. Your proxy's remote server is out of service.

      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/boost-cmake/portfile.cmake:5 (vcpkg_from_github)
  scripts/ports.cmake:192 (include)

error: building boost-cmake:x64-windows failed with: BUILD_FAILEDSee https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle boost-cmake:x64-windows: 90.9 ms
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+boost-cmake
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[boost-cmake]+Build+error+on+x64-windows&body=Copy+issue+body+from+E%3A%2FC%E8%AF%AD%E8%A8%80%2F3D_mesh%2FProject1%2Fvcpkg%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

**Comment by FrankXie05** [comment]
This link `https://github.com/boostorg/boost/archive/boost-1.85.0.tar.gz ` is valid. Please make sure your network is connected and proxy is enabled through vcpkg.

**Comment by newne** [comment]
the same problem. Here is the console output:
```
(base) PS E:\code\vcpkg> .\vcpkg.exe install cgal
Computing installation plan...
The following packages will be built and installed:
  * boost-format:x64-windows@1.85.0#1
  * boost-geometry:x64-windows@1.85.0#2
  * boost-graph:x64-windows@1.85.0#1
  * boost-heap:x64-windows@1.85.0#1
  * boost-iostreams[bzip2,core,lzma,zlib,zstd]:x64-windows@1.85.0#1
  * boost-math:x64-windows@1.85.0#2
  * boost-multi-array:x64-windows@1.85.0#1
  * boost-multiprecision:x64-windows@1.85.0#2
  * boost-polygon:x64-windows@1.85.0#1
  * boost-property-map:x64-windows@1.85.0#1
  * boost-ptr-container:x64-windows@1.85.0#1
  * boost-qvm:x64-windows@1.85.0#1
  * boost-rational:x64-windows@1.85.0#1
  * boost-units:x64-windows@1.85.0#1
  * boost-xpressive:x64-windows@1.85.0#1
  * bzip2[core,tool]:x64-windows@1.0.8#5
    cgal:x64-windows@5.6.1#1
  * gmp:x64-windows@6.3.0
  * liblzma:x64-windows@5.6.2
  * mpfr:x64-windows@4.2.1
  * vcpkg-cmake-get-vars:x64-windows@2023-12-31
  * zlib:x64-windows@1.3.1
  * zstd:x64-windows@1.5.6
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
Compiler found: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\ncgne\AppData\Local\vcpkg\archives in 15.9 ms. Use --debug to see more details.
Installing 1/23 boost-format:x64-windows@1.85.0#1...
Building boost-format:x64-windows@1.85.0#1...
-- Downloading https://github.com/boostorg/format/archive/boost-1.85.0.tar.gz -> boostorg-format-boost-1.85.0.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from E:\code\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: E:\code\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
warning: Download failed -- retrying after 1000ms
warning: Download failed -- retrying after 2000ms
warning: Download failed -- retrying after 4000ms
error: Missing boostorg-format-boost-1.85.0.tar.gz and downloads are blocked by x-block-origin.
error: https://github.com/boostorg/format/archive/boost-1.85.0.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/boostorg/format/archive/boost-1.85.0.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/boostorg/format/archive/boost-1.85.0.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/boostorg/format/archive/boost-1.85.0.tar.gz: WinHttpSendRequest failed with exit code 12029
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030):
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 3us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 5744us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 15 s (15026172us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):

      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.

      3. Your proxy's remote server is out of service.

      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.


Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/boost-format/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:192 (include)


error: building boost-format:x64-windows failed with: BUILD_FAILEDSee https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle boost-format:x64-windows: 15 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+boost-format
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[boost-format]+Build+error+on+x64-windows&body=Copy+issue+body+from+E%3A%2Fcode%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```

**Comment by MrLyagushka** [comment]
Попробуй переименовать названия папок, в которых лежит библиотека. Мне это помогло.
 Try renaming the names of the folders in which the library is located. It helped me.

**Comment by lukasblass** [comment]
I am experiencing the same issue, does anyone have a solution to this?

**Comment by JonLiu1993** [comment]
Duplicate of #41199
